float64 move
int32 rotation_right
int32 rotation_left
